### PR TITLE
Match types for User#callFunction implementation across platforms

### LINKED ISSUE #3733: Types for `User#callFunction` doesn't match implementation across platforms
## Goals

A strongly typed and unified API across Realm JS and Realm Web.

## Expected Results

I would expect calling a MongoDB Realm function would have the same signature across our supported platforms.

## Actual Results

The [TypeScript declarations](https://github.com/realm/realm-js/blob/master/types/app.d.ts#L495) and [Realm Web implementation](https://github.com/realm/realm-js/blob/master/packages/realm-web/src/User.ts#L322-L324) takes arguments using ["rest parameter"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) syntax. Whereas the Realm JS implemeation and [JS docs](https://github.com/realm/realm-js/blob/master/docs/sync.js#L698-L704) takes an array of arguments as its 2nd positional argument.

We might consider deprecating and consider the `callFunction` method private altogether, since we intent user to call functions in the slightly more convenient syntax:

```
user.functions.sum(1, 2);
```

## Version of Realm and Tooling

- Realm JS SDK Version: Realm JS 10.4.0 and Realm Web 1.2.1
- Node, React Native and Web.
- Client OS & Version: N/A
- Which debugger for React Native: None

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
The unifies the call signature of `User#callFunction` across all documentation
and platforms.

This closes #3733 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Review by kneth — APPROVED** [review]
LGTM. But if it is only documentation changes, it is a breaking change?

File: types/index.d.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ declare namespace Realm {
      * PropertyType
      * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~PropertyType }
      */
-    type PropertyType = string | 'bool' | 'int' | 'float' | 'double' | 'decimal128' | 'objectId' | 'string' | 'data' | 'date' | 'list' | 'linkingObjects';
+    type PropertyType = 'bool' | 'int' | 'float' | 'double' | 'decimal128' | 'objectId' | 'string' | 'data' | 'date' | 'list' | 'linkingObjects';
```
Comment (takameyer) [inline types/index.d.ts:39]: Fix this

**Comment by takameyer** [comment]
@kneth Went over this with @kraenhansen.  This is a breaking change,  as it changes the `callFunction` implementation in `realm-js`:
```
    user.callFunction("sum", 1, 2, 3); // Valid
    user.callFunction("sum", [1, 2, 3]); // Invalid
```

File: lib/user.js [hunk 2]
```diff
@@ -98,15 +102,15 @@ const instanceMethods = {
...
+  _functionsOnService(serviceName) {
     const user = this;
     return new Proxy(
       {},
       {
         get(target, name, receiver) {
           if (typeof name === "string" && name != "inspect") {
             return (...args) => {
-              return user.callFunction(name, args, service);
+              return user._callFunctionOnService(name, serviceName, args);
```
Comment (kraenhansen) [inline lib/user.js:113]: I think these arguments should be spread into the call.

```suggestion
              return user._callFunctionOnService(name, serviceName, ...args);
```

Comment (takameyer) [reply lib/user.js]: Good catch.  We also needed to return the call to  `_callFunctionOnService`.  Jenkins tests should pass now

**Pull request merged**